### PR TITLE
Added command option parsing, and --model, --anchors, --classes, and …

### PR DESCRIPTION
**Pull request by philtrade**
Added command option parsing.

User can specify model/weight file, anchors file, classes file, and # of GPU to use, via --model <file>, --anchors <file>, --classes <file>, and --gpu_num N.   Use previous hardcoded values as default.

### CODE REVIEW

**Comment by tanakataiki** [comment]
And there is 2 command line input with sys.argv, so can you move it to yolo_video and add to it with detect_image ?

**Comment by philtrade** [comment]
Will look into yolo_video later, and work out a separate patch/pull
request.  Thx for the suggestion.

**Comment by tanakataiki** [comment]
I think needs for image is lower than video, and I dont use detection for image so mainly yolo_video would be Run file. Currently I dont meet some trouble this this code so smaller change for better hack should be appreciated. Also, I think run file must be summarized in One file and should be apart from setting or Network construction file. So I meant "moving detect_img(YOLO()) out of yolo.py, into yolo_video.py", and add the command line options in yolo_video.py to detect video or image decided by return value would be better. 
Note The change shouldnt affect the way to use of previous commit yolo_video.py... ideally.
And can you summarize on a few commit so that we can tell whats changed?

**Comment by philtrade** [comment]
Hi Taiki,

Okay I will refactor the code more like this, correct me if I'm wrong:

- both detect_img() and detect_video() should be moved out of yolo.py to
yolo_video.py.

- add command line option in yolo_video.py, but may need to change the
semantics a little bit.  How about:

  for image detection: yolo_video.py  --image <optional image files.  If
not provided, let user input filename in the interactive mode>

  for video detection:  yolo_video.py  video_input_path  <optional output
path>

In my humble opinion, if a single run file is used, it maybe renamed from
"yolo_video.py" to something like "yolo_detect.py".  But that will break
the old semantics.

What do you think Taiki?
